apps/backend/src/app/modules/killswitch/killswitch.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Inject, Injectable, OnModuleInit } from '@nestjs/common';
+import { EXTENDED_PRISMA_SERVICE } from '../database/db.constants';
+import { ExtendedPrismaService } from '../database/prisma.extension';
+import { KillswitchType } from './killswitch.enum';
+import { Enum } from '@momentum/enum';
+
+export type Killswitches = Record<KillswitchType, boolean>;
+export const KILLSWITCH_CONFIG_ID = 'Killswitches';
+
+@Injectable()
+export class KillswitchService implements OnModuleInit {
+  constructor(
+    @Inject(EXTENDED_PRISMA_SERVICE) private readonly db: ExtendedPrismaService
+  ) {}
+
+  private switches: Killswitches;
+
+  public async onModuleInit() {
+    const dbResponse = await this.db.config.findUnique({
+      where: { id: KILLSWITCH_CONFIG_ID }
+    });
+
+    const storedSwitches = dbResponse?.value as Killswitches;
+    if (storedSwitches) {
+      this.validateKillswitches(storedSwitches);
+      this.switches = storedSwitches;
+    } else {
+      this.switches = {} as Killswitches;
+      await this.updateKillswitches(
+        Object.fromEntries(
+          Enum.values(KillswitchType).map((type: KillswitchType) => [
+            type,
+            false
+          ])
+        ) as Killswitches
+      );
+    }
```

Comment:
Maybe worth adding some logging in here. If the entry doesn't exist, let's log that we created it. If it *does* exist, and certain killswitches are enabled, let's output a log message (info level) for all the enabled switches.

Reply:
So create a new logger instance in the service?

Reply:
Yeah, though tbh I don't remember exactly how we do it, have a look round the codebase